torch/_inductor/codegen/cpp.py

```diff
@@ -2247,19 +2276,22 @@ def vec_to_array(vec_var: CppCSEVariable) -> CppCSEVariable:
     def load(self, name: str, index: sympy.Expr):
         opt_ctx: OptimizationContext = get_current_node_opt_ctx()
         var = self.args.input(name)
+        original_index = index
         index = self.rename_indexing(index)
         dtype = V.graph.get_dtype(name)
         tiling_var = self.itervars[self.tiling_idx]
         stride = self._try_get_const_stride(index, tiling_var)
         if stride == 0:
             # load scalar and lazily broadcast it on demand
             return super().load(name, index)
-        elif stride == 1:
-            # load contiguously
-            line = self._get_vec_load_line(var, index, dtype, self._load_mask)
-            csevar = self.cse.generate(self.loads, line)  # type: ignore[assignment]
         else:
-            csevar = self._load_or_store_non_contiguous(var, index, dtype)  # type: ignore[assignment]
+            # Issue necessary TORCH_CHECKs for the indirect indexing
+            if stride == 1:
+                # load contiguously
+                line = self._get_vec_load_line(var, index, dtype, self._load_mask)
+                csevar = self.cse.generate(self.loads, line)  # type: ignore[assignment]
+            else:
+                csevar = self._load_or_store_non_contiguous(var, original_index, dtype)  # type: ignore[assignment]
```

Comment:
I think most of the changes in this file are stale, except for adding `check_bounds`.